src/components/Swap/SwapActionButton/SwapButton.tsx

```diff
@@ -19,7 +19,7 @@ import invariant from 'tiny-invariant'
 import ActionButton from '../../ActionButton'
 import Dialog from '../../Dialog'
 import { SummaryDialog } from '../Summary'
-import { useCollapseToolbar } from '../Toolbar'
+import { useCollapseToolbar } from '../Toolbar/ToolbarContext'
```

Comment:
Instead of importing this, which although is not technically circular still seems circular (Toolbar iimport SwapButton imports Toolbar/ToolbarContext), could you pass the collapse toolbar callback as a prop (`onBeforeReviewSwap`?) to keep these components more distinct?

I'm also fine with this as a fast follow-up, so that you can implement a minimal change to fix the build _first_, and _then_ clean up the implementation.

Reply:
i see your point but want to discuss this before making the change, so i'll do it in a follow up PR